### ISSUE TITLE
Stash only selected files in `Unstaged` area

### ISSUE THREAD

**Issue by feeeper** [ForkIssues/TrackerWin#1375]
Not a bug (I think) but really annoying behaviour:
1. I changed lot of files in a project
2. I selected some of changed files in the `Unstaged` area
3. I click "Stash" button on the panel

Expecting result:
1. Only selected files will be stashed

Actual result:
1. By default all unstaged files will be stashed

**Comment by danieldownes** [comment]
You can now stash staged files, rather than all files. So just select and stage files, then stash.

Issue solved.

**Comment by feeeper** [comment]
@danieldownes I think that I describe steps for reporduce not very clear. I've attached GIF what I do:
![gif](https://i.gyazo.com/42b10ad7965ce2d560b3381315925318.gif)

In this case I expect that only selected files will be stashed. I expect that behaviour will be the same as I stash files via right click:

![img](https://i.gyazo.com/25a9c0d3ee57238c4c357ed3810a75dd.png)

**Comment by michael-hawker** [comment]
I'm also still seeing weird behavior here. I select 2 files to stash from Unstaged (I have two other files in staged). I see in the dialog only the two files I selected checked. Then I confirm, but then my stash contains all 4 files, the two I wanted and the 2 staged ones...

**Comment by DanPristupov** [comment]
Not sure if I'm missing something. This is how git works, I think.

Fork uses `git stash --push -- path/to/file1 path/to/file2` command.

`git stash` docs (https://git-scm.com/docs/git-stash#Documentation/git-stash.txt-ltpathspecgt82308203):

> `<pathspec>`
> 
> The new stash entry records the modified states only for the files that match the pathspec. The index entries and working tree files are then rolled back to the state in HEAD only for these files, too, leaving files that do not match the pathspec intact.

**Comment by feeeper** [comment]
> Fork uses `git stash --push -- path/to/file1 path/to/file2` command.

@DanPristupov Yes, I get it. I expected that only selected files in the **Unstaged** section would be stashed if any. In case there are no selected files in that area then all files would be stashed.

It would be fine (for me obviously) if the dialog **Save stash** will appear when I click **Stash** button on the panel with only selected files "checked".
![image](https://github.com/fork-dev/TrackerWin/assets/3373303/348b145c-9c0f-4191-b135-8e93c7db009b)
